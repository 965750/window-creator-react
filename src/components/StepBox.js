import React from 'react'
import styles from './StepBox.module.scss'
import { connect } from 'react-redux'
import { changeStep } from '../store/actions/themeActions'

function StepBox ({ step, active, handleChangeStep }) {
    return (
        <div className='w-1/3 text-center z-20'>
            { active }
            <div onClick={() => handleChangeStep(step.id)} className={`${styles.circle} ${active ? 'bg-gullGray' : 'bg-botticelli'} cursor-pointer rounded-full bg-botticelli relative mx-auto`}>
                <div className={`${styles.circle__inside} ${active ? 'bg-white' : 'bg-hippieBlue'} top-half left-half absolute rounded-full`} />
            </div>
            <div onClick={() => handleChangeStep(step.id)}>
                <p className={`${styles.circle__text} uppercase font-bold mt-3`}>
                    {step.label}
                </p>
                <p className={`${styles.circle__text} uppercase font-bold`}>
                    {step.title}
                </p>
            </div>
        </div>
    )
}

const mapDispatchToProps = (dispatch) => {
    return {
        handleChangeStep: (step) => dispatch(changeStep(step))
    }
}

export default connect(null, mapDispatchToProps)(StepBox)
