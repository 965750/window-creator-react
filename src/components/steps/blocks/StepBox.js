import React from 'react'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import PropTypes from 'prop-types'
import { changeStep } from '../../../store/actions/themeActions'
import styles from './StepBox.module.scss'

function StepBox({ step, active, handleChangeStep }) {
    return (
        <div className="w-1/3 text-center z-20">
            {active}
            <div
                onClick={() => handleChangeStep(step.id)}
                className={`${styles.circle} ${
                    active ? 'bg-gullGray' : 'bg-botticelli'
                } cursor-pointer rounded-full bg-botticelli relative mx-auto`}
            >
                <div
                    className={`${styles.circle__inside} ${
                        active ? 'bg-white' : 'bg-hippieBlue'
                    } top-half left-half absolute rounded-full`}
                />
            </div>
            <div onClick={() => handleChangeStep(step.id)}>
                <p
                    className={`${styles.circle__text} uppercase font-bold mt-3`}
                >
                    <FormattedMessage
                        id={step.label}
                        defaultMessage={step.label}
                    />
                </p>
                <p className={`${styles.circle__text} uppercase font-bold`}>
                    <FormattedMessage
                        id={step.title}
                        defaultMessage={step.title}
                    />
                </p>
            </div>
        </div>
    )
}

const mapDispatchToProps = dispatch => ({
    handleChangeStep: step => dispatch(changeStep(step)),
})

StepBox.propTypes = {
    step: PropTypes.object.isRequired,
    active: PropTypes.bool,
    handleChangeStep: PropTypes.func,
}

export default connect(
    null,
    mapDispatchToProps
)(StepBox)
