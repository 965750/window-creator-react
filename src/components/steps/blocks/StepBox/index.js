import { changeStep } from '../../../../store/actions/themeActions'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'

import PropTypes from 'prop-types'
import React from 'react'
import styles from './StepBox.module.scss'

function StepBox({ step, active, handleChangeStep }) {
  return (
    <div className="w-1/3 text-center z-20">
      <div
        data-testid="stepBoxCircle"
        onClick={() => handleChangeStep(step.id)}
        className={`${styles.circle} ${
          active ? 'bg-gullGray' : 'bg-botticelli'
        } cursor-pointer rounded-full bg-botticelli relative mx-auto`}
      >
        <div
          data-testid="stepBoxDot"
          className={`${styles.circle__inside} ${
            active ? 'bg-white' : 'bg-hippieBlue'
          } top-half left-half absolute rounded-full`}
        />
      </div>
      <div onClick={() => handleChangeStep(step.id)}>
        <p className={`${styles.circle__text} uppercase font-bold mt-3`}>
          <FormattedMessage id={step.label} defaultMessage={step.label} />
        </p>
        <p className={`${styles.circle__text} uppercase font-bold`}>
          <FormattedMessage id={step.title} defaultMessage={step.title} />
        </p>
      </div>
    </div>
  )
}

const mapDispatchToProps = (dispatch) => ({
  handleChangeStep: (step) => dispatch(changeStep(step)),
})

StepBox.propTypes = {
  step: PropTypes.shape({
    title: PropTypes.string,
    label: PropTypes.string,
    id: PropTypes.number,
  }),
  active: PropTypes.bool,
  handleChangeStep: PropTypes.func,
}

export default connect(
  null,
  mapDispatchToProps,
)(StepBox)
