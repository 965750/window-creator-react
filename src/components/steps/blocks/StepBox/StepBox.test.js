import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import {render, cleanup, fireEvent} from '@testing-library/react';

import StepBox from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
  creator: {
    stepper: {
      active: 1
    }
  }
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    case 'CHANGE_STEP':
      return {
        ...state,
        creator: {
          stepper: {
            active: action.step
          }
        }
      }
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('StepBox', () => {
  let wrapper
  const initialProps = {
    step: {
      title: 'mockTitle',
      label: 'mockLabel',
      id: 7,
    },
    active: false,
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <StepBox {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should have gullGray background and white dot inside when step is active', () => {
    setUp({
      ...initialProps,
      active: true
    })

    expect(wrapper.queryByTestId('stepBoxCircle')).toHaveClass('bg-gullGray')
    expect(wrapper.queryByTestId('stepBoxDot')).toHaveClass('bg-white')
  })

  it('should call dispatch action to change step', () => {
    setUp()

    expect(store.getState().creator.stepper.active).toBe(1)

    fireEvent.click(wrapper.queryByTestId('stepBoxCircle'))

    expect(store.getState().creator.stepper.active).toBe(7)
  })
})
