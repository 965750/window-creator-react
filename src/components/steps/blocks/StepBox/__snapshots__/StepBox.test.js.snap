// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepBox should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-1/3 text-center z-20"
  >
    <div
      class="circle bg-botticelli cursor-pointer rounded-full bg-botticelli relative mx-auto"
      data-testid="stepBoxCircle"
    >
      <div
        class="circle__inside bg-hippieBlue top-half left-half absolute rounded-full"
        data-testid="stepBoxDot"
      />
    </div>
    <div>
      <p
        class="circle__text uppercase font-bold mt-3"
      >
        mockLabel
      </p>
      <p
        class="circle__text uppercase font-bold"
      >
        mockTitle
      </p>
    </div>
  </div>
</DocumentFragment>
`;
